universal-application-tool-0.0.1/app/views/admin/programs/ProgramFormBuilder.java

```diff
@@ -13,6 +15,25 @@
  */
 public class ProgramFormBuilder extends BaseHtmlView {
 
+  public static ContainerTag buildProgramForm(ProgramForm program, boolean editExistingProgram) {
```

Comment:
(non-blocking): I'm not super familiar with this class. If it's not too much trouble, can you add a short docstring to each of these `buildProgramForm` methods explaining when you'd want to use each of them? It's not clear to me why we need so many different public method signatures at first glance.

Reply:
I'm not super familiar with this either. It looks like you can build the form using data either from a `ProgramForm` or a `ProgramDefinition`. The former is passed when a user updates a program and stays on the page (and may include update errors). The latter is used when reading data from the database.

It's mainly just me relocating the mapping from `ProgramEditView` and `ProgramNewOneView` into `ProgramFormBuilder`

Reply:
But does the the third version below these two need to be public? Or is it only ever called from within this class?

Reply:
The third method should be private now.